boa3_test/tests/compiler_tests/test_builtin_method.py

```diff
@@ -1010,3 +1010,135 @@ def test_str_split_default(self):
         self.assertEqual(expected_result, result)
 
     # endregion
+
+    # region count test
+
+    def test_count_list_int(self):
+        path = self.get_contract_path('CountListInt.py')
+        engine = TestEngine()
+
+        list_ = [1, 2, 3, 4, 1, 1, 0]
+        expected_result = list_.count(1)
+        result = self.run_smart_contract(engine, path, 'main')
+        self.assertEqual(expected_result, result)
+
+    def test_count_list_str(self):
+        path = self.get_contract_path('CountListStr.py')
+        engine = TestEngine()
+
+        list_ = ['unit', 'test', 'unit', 'unit', 'random', 'string']
+        expected_result = list_.count('unit')
+        result = self.run_smart_contract(engine, path, 'main')
+        self.assertEqual(expected_result, result)
+
+    def test_count_list_bytes(self):
+        path = self.get_contract_path('CountListBytes.py')
+        engine = TestEngine()
+
+        list_ = [b'unit', b'test', b'unit', b'unit', b'random', b'string']
+        expected_result = list_.count(b'unit')
+        result = self.run_smart_contract(engine, path, 'main')
+        self.assertEqual(expected_result, result)
+
+    def test_count_list_different_types(self):
+        path = self.get_contract_path('CountListDifferentTypes.py')
+        engine = TestEngine()
+
+        list_ = [b'unit', 'test', b'unit', b'unit', 123, 123, True, False]
+        expected_result = list_.count(b'unit'), list_.count('test'), list_.count(123)
+        result = self.run_smart_contract(engine, path, 'main')
+        self.assertEqual(expected_result, tuple(result))
+
+    def test_count_list_different_non_primitive_types(self):
+        path = self.get_contract_path('CountListDifferentNonPrimitiveTypes.py')
+        engine = TestEngine()
+
+        list_ = [[b'unit', 'test'], [b'unit', b'unit'], [123, 123], [True, False], [True, False]]
+        expected_result = list_.count([b'unit', 'test']), list_.count([123, 123]), list_.count([True, False])
+
+        # TODO: change test when comparison of non primitive types are implemented in the Opcode
+        result = self.run_smart_contract(engine, path, 'main')
+
+        with self.assertRaises(AssertionError):
+            self.assertEqual(expected_result, tuple(result))
```

Comment:
Instead of expecting an error in this assertion, use the proper assert method
```suggestion
        self.assertNotEqual(expected_result, tuple(result))
```

Reply:
But according to your comment in the code, this is a case that is not implemented right?
In this case, the expected behavior should be the compiler raising an `NotSupportedException` and failing to compile.